drivers/sensor/MLX90640.zig

```diff
@@ -0,0 +1,896 @@
+
+        self.params.cpAlpha[0] = alphaSP[0];
+        self.params.cpAlpha[1] = alphaSP[1];
+        self.params.cpOffset[0] = offsetSP[0];
+        self.params.cpOffset[1] = offsetSP[1];
+    }
+
+    fn extract_deviating_pixels(self: *Self) i16 {
+        var pixCnt: u32 = 0;
+        for (0..5) |i| {
+            pixCnt = @intCast(i);
+            self.params.brokenPixels[pixCnt] = 0xFFFF;
+            self.params.outlierPixels[pixCnt] = 0xFFFF;
+        }
+
+        var brokenPixCnt: u16 = 0;
+        var outlierPixCnt: u16 = 0;
+        pixCnt = 0;
+        while (pixCnt < 768 and brokenPixCnt < 5 and outlierPixCnt < 5) {
+            if (self.eeprom[pixCnt + 64] == 0) {
+                self.params.brokenPixels[brokenPixCnt] = @intCast(pixCnt);
+                brokenPixCnt = brokenPixCnt + 1;
+            } else if ((self.eeprom[pixCnt + 64] & 0x0001) != 0) {
+                self.params.outlierPixels[outlierPixCnt] = @intCast(pixCnt);
+                outlierPixCnt = outlierPixCnt + 1;
+            }
+
+            pixCnt = pixCnt + 1;
+        }
+
+        var warn: i16 = 0;
+        if (brokenPixCnt > 4) {
+            warn = -3;
+        } else if (outlierPixCnt > 4) {
+            warn = -4;
+        } else if ((brokenPixCnt + outlierPixCnt) > 4) {
+            warn = -5;
+        } else {
+            for (0..brokenPixCnt) |x| {
+                pixCnt = @intCast(x);
+                for (pixCnt + 1..brokenPixCnt) |i| {
+                    warn = self.check_adjacent_pixels(self.params.brokenPixels[pixCnt], self.params.brokenPixels[i]);
+                    if (warn != 0) {
+                        return warn;
+                    }
+                }
+            }
+
+            for (0..outlierPixCnt) |x| {
+                pixCnt = @intCast(x);
+                for (pixCnt + 1..outlierPixCnt) |i| {
+                    warn = self.check_adjacent_pixels(self.params.outlierPixels[pixCnt], self.params.outlierPixels[i]);
+                    if (warn != 0) {
+                        return warn;
+                    }
+                }
+            }
+
+            for (0..brokenPixCnt) |x| {
+                pixCnt = @intCast(x);
+                for (0..outlierPixCnt) |i| {
+                    warn = self.check_adjacent_pixels(self.params.brokenPixels[pixCnt], self.params.outlierPixels[i]);
+                    if (warn != 0) {
+                        return warn;
+                    }
+                }
+            }
+        }
+
+        return warn;
+    }
+
+    fn check_adjacent_pixels(_: *Self, pix1: u16, pix2: u16) i16 {
+        var pixPosDif: i32 = 0;
+
+        pixPosDif = pix1 - pix2;
+        if (pixPosDif > -34 and pixPosDif < -30) {
+            return -6;
+        }
+        if (pixPosDif > -2 and pixPosDif < 2) {
+            return -6;
+        }
+        if (pixPosDif > 30 and pixPosDif < 34) {
+            return -6;
+        }
+
+        return 0;
+    }
+};
+
+test "refresh rate" {
```

Comment:
When I run zig test I get 
```drivers/sensor/MLX90640.zig:12:22: error: import of file outside module path: '../framework.zig'
const base = @import("../framework.zig").base;```

I can run other tests, I'm not sure why mine is different.

Reply:
Aha!  The tests should be in the MLX90640 struct.  I don't really understand why, but I got them working.

Reply:
You figured this out, right?

Reply:
Yes, I can run the tests now.